modules/templates-writing-parameters.adoc

```diff
@@ -63,13 +63,19 @@ characters long consisting of all upper and lowercase alphabet letters
 and numbers.
 
 The syntax available is not a full regular expression syntax. However, you can
-use `\w`, `\d`, and `\a` modifiers:
+use `\\w`, `\\d`, `\\a`, and `\\A` modifiers:
 
-- `[\w]{10}` produces 10 alphabet characters, numbers, and underscores. This
+- `[\\w]{10}` produces 10 alphabet characters, numbers, and underscores. This
 follows the PCRE standard and is equal to `[a-zA-Z0-9_]{10}`.
-- `[\d]{10}` produces 10 numbers. This is equal to `[0-9]{10}`.
-- `[\a]{10}` produces 10 alphabetical characters. This is equal to
+- `[\\d]{10}` produces 10 numbers. This is equal to `[0-9]{10}`.
+- `[\\a]{10}` produces 10 alphabetical characters. This is equal to
 `[a-zA-Z]{10}`.
+- `[\\A]{10}` produces 10 punctuation or symbol characters. This is equal to `[~!@#$%\^&*()\-_+={}\[\]\\|<,>.?/"';:]{10}`.
+
+[NOTE]
+====
+In the example, the `\`, `^`, `-`, `[`, and `]` characters are used as escape characters. The `\\A` parameter would be any of the following characters: `~!@#$%^&*()-_+={}[]\|<,>.?/"';:`.
```

Comment:
> Actually I think this wording would be better:
> 
> The `\\A` parameter expands to one of the following characters: `` ~!@#$%^&*()-_+={}[]\|<,>.?/"';:` ``. In the example regular expression, a backslash is used to disable the special meaning of the `\`, `^`, `-`, `[` and `]` characters.

@EricPonvelle @yrro Is it necessary to escape the special characters in the [\A]{10} bullet example above? Or can the bullet be the following with a note afterwards explaining the need to escape the special characters:

> * `[\A]{10}` produces 10 punctuation or symbol characters. This is equal to `[~!@#$%^&*()-_+={}[]\|<,>.?/"';:]{10}`.
> 
> **NOTE**
> In a regular expression, the special meaning of the `\`, `^`, `-`, `[`, and `]` characters can be escaped with a backslash. If the template is written in JSON or YAML, and depending on the type of string that it is embedded within, the backslash in `[\A]` might also need to be escaped.

Reply:
@xiuwang can you review Lindsey's question as well?

Reply:
According to [Special Characters Inside a Bracketed Character Class](https://perldoc.perl.org/perlrecharclass#Special-Characters-Inside-a-Bracketed-Character-Class):

> Characters that may carry a special meaning inside a character class are: `\`, `^`, `-`, `[` and `]`, and are discussed below. They can be escaped with a backslash, although this is sometimes not needed, in which case the backslash may be omitted.

So you either have to play games with the positioning of the characters within the bracketed character class (don't put `^` at the start; do put `-` at the start; don't put a `[` where it might accidently construct a POSIX character class; don't put a `\` in front of three numbers or certain letters... probably others...)

... or just escape the lot of them and not worry about any of it. :)

TBH maybe it's wroth re-assessing the value of providing the definition of `\A` in terms of a PCRE at all. For `\w`, `\d` and `\a` the PCREs can be taken in at a glance and so are quite useful. But for `\W`, however, I think the fact that we are having to fight with our own understanding of the ins and outs of PCRE syntax (not to mention the Markdown escaping!) demonstrates that documenting `\A` in terms of a PCRE is more trouble than it's worth. In the current proposed change, I think the average user will skip straight to the explanatory note rather than referring to the PCRE definition.